.github/workflows/sycl-linux-run-tests.yml

```diff
@@ -312,178 +291,26 @@ jobs:
-      uses: actions/upload-artifact@v4
+      uses: ./devops/actions/run-tests/e2e
       with:
-        name: sycl_cts_bin
-        path: sycl_cts_bin.tar.zst
+        ref: ${{ inputs.ref || github.sha }}
+        merge_ref: ${{ inputs.merge_ref }}
+        e2e_binaries_artifact: ${{ inputs.e2e_binaries_artifact }}
+        extra_cmake_args: ${{ inputs.extra_cmake_args }}
+        e2e_testing_mode: ${{ inputs.e2e_testing_mode }}
+        target_devices: ${{ inputs.target_devices }}
+        extra_lit_opts: ${{ inputs.extra_lit_opts }}
+        artifact_suffix: ${{ inputs.artifact_suffix }}
         retention-days: ${{ inputs.retention-days }}
 
-    - name: Download SYCL-CTS binaries
-      if: inputs.sycl_cts_artifact != ''
-      uses: actions/download-artifact@v4
-      with:
-        name: ${{ inputs.sycl_cts_artifact }}
-
-    - name: Extract SYCL-CTS binaries
-      if: inputs.sycl_cts_artifact != ''
-      run: |
-        mkdir -p build-cts/bin
-        tar -I 'zstd' -xf sycl_cts_bin.tar.zst -C build-cts/bin
-
-    - name: SYCL CTS List devices
-      # Proceed with execution even if the 'build' step did not succeed.
-      if: inputs.tests_selector == 'cts' && (always() && !cancelled()) && inputs.cts_testing_mode != 'build-only'
-      env:
-        ONEAPI_DEVICE_SELECTOR: ${{ inputs.target_devices }}
-      run: |
-        ./build-cts/bin/* --list-devices
-
-    # If the suite was built on another machine then the build contains the full
-    # set of tests. We have special files to filter out some test categories,
-    # see "devops/cts_exclude_filter_*". Each configuration has its own file, e.g.
-    # there is "cts_exclude_filter_OCL_CPU" for opencl:cpu device. Therefore,
-    # these files may differ from each other, so when there is a pre-built set of
-    # tests, we need to filter it according to the filter-file.
-    - name: Filter SYCL CTS test categories
-      if: inputs.sycl_cts_artifact != ''
-      shell: bash
-      run: |
-        cts_exclude_filter=""
-        if [ "${{ contains(inputs.target_devices, 'opencl:cpu')  }}" = "true" ]; then
-          cts_exclude_filter=$PWD/devops/cts_exclude_filter_OCL_CPU
-        elif [ "${{ contains(inputs.target_devices, 'level_zero:gpu')  }}" = "true" ]; then
-          cts_exclude_filter=$PWD/devops/cts_exclude_filter_L0_GPU
-        fi
-
-        while IFS= read -r line; do
-          if [[ $line != \#* ]]; then
-            rm "./build-cts/bin/test_$line"
-          fi
-        done < "$cts_exclude_filter"
-
-    - name: Run SYCL CTS tests
-      # Proceed with execution even if the previous two steps did not succeed.
-      if: inputs.tests_selector == 'cts' && (always() && !cancelled()) && inputs.cts_testing_mode != 'build-only'
-      env:
-        ONEAPI_DEVICE_SELECTOR: ${{ inputs.target_devices }}
-      # This job takes ~100min usually. But sometimes some test isn't
-      # responding, so the job reaches the 360min limit. Setting a lower one.
-      timeout-minutes: 150
```

Comment:
I couldn't preserve this for whatever weird GHA reasons, hope that's fine for now. We can try to introduce another mitigation later if necessary.

Reply:
It was made due to weird machine that was hanging sometimes. No need to re-introduce.